internal/channel/plugin.go

```diff
@@ -27,60 +83,102 @@ func (c *Channel) GetInfo() (*plugin.Info, error) {
 	return info, nil
 }
 
-func (c *Channel) SetConfig(config string) error {
-	_, err := c.rpcCall(plugin.MethodSetConfig, json.RawMessage(config))
+// SetConfig sends the setConfig request with given config, returns an error if an error occurred
+func (p *Plugin) SetConfig(config string) error {
+	_, err := p.rpc.Call(plugin.MethodSetConfig, json.RawMessage(config))
 
 	return err
 }
 
-func (c *Channel) SendNotification(contact *recipient.Contact, i contracts.Incident, ev *event.Event, icingaweb2Url string) error {
-	contactStruct := &plugin.Contact{FullName: contact.FullName}
-	for _, addr := range contact.Addresses {
-		contactStruct.Addresses = append(contactStruct.Addresses, &plugin.Address{Type: addr.Type, Address: addr.Address})
+// SendNotification sends the NotificationRequest as a go routine
+func (p *Plugin) SendNotification(req *plugin.NotificationRequest) {
```

Comment:
> as a go routine

This one doesn't even start one, but it sounds like it did, while it's waiting for the response.